### PR TITLE
Feature/mpi

### PR DESCRIPTION
**Pull request by bartnijssen**
Don't accept - this is just to visualize the changes that were made to support MPI

### CODE REVIEW

File: drivers/image/src/vic_start.c [hunk 1]
```diff
@@ -48,33 +62,127 @@ vic_start(void)
...
-    get_global_param(filep.globalparam);
+    if (mpi_rank == 0) {
+        // read global settings
+        filep.globalparam = open_file(filenames.global, "r");
+        get_global_param(filep.globalparam);
+    }
 
     // Set Log Destination
+    // TBD: Add MPI support to logging: every process should log to its own file
     setup_logging();
```
Comment (jhamman) [inline drivers/image/src/vic_start.c]: This should be easy enough to do.  I think we just need to pass the `mpi_rank` as an argument to [`setup_logging()`](https://github.com/UW-Hydro/VIC/blob/develop/drivers/shared/src/vic_log.c#L116) to help define the log file names.

**Comment by jhamman** [comment]
Nice work Bart.  A few questions...
- Where are the number of mpi tasks defined?  Is this a runtime option?
- Obviously, we'll need to run some tests to check the model outputs, performance, etc.  Are there additional features or commits that will be needed before we get to that point?

I'll can convert the Stehekin test case to netCDF format allowing us to use that as a test case.

**Comment by bartnijssen** [comment]
The number of tasks is defined at run-time, for example using mpich, you'd start the run with something along the lines of:

    mpirun-mpich-mp -np 4 vic_image -g global_file

You can also just run as 

    vic_image -g global_file

which will be a single process run.

All the configuration with respect to number of nodes and number of processes per node is done outside of VIC. The VIC code just knows about the number of processes as `mpi_size` and the rank of any individual process as `mpi_rank` (with the master process equal to `mpi_rank == 0`).

I think that other than the logging (which I'll update), this should be ready for testing. It probably won't run at first, since I haven't had a good dataset to debug with, so I'll probably need to do a little bit more work before I can hand it off and merge it with the main `develop` branch.

As for performance issues: I know the NetCDF IO is currently slow, probably at least in part because I may be making too many queries (I should be able to store the info on a file, keep it open and then read from or write to it). I think these can be worked on once we have a code that we know gets the correct answer.

**Comment by bartnijssen** [comment]
BTW converting the Stehekin case would be great as we can run that for older versions of VIC as well.

**Comment by bartnijssen** [comment]
Closing this for now

**Pull request closed without merging**